src/ModelContextProtocol/Configuration/McpServerServiceCollectionExtensions.cs

```diff
@@ -17,7 +18,7 @@ public static class McpServerServiceCollectionExtensions
     public static IMcpServerBuilder AddMcpServer(this IServiceCollection services, Action<McpServerOptions>? configureOptions = null)
     {
         services.AddOptions();
-        services.AddTransient<IConfigureOptions<McpServerOptions>, McpServerOptionsSetup>();
+        services.TryAddEnumerable(ServiceDescriptor.Transient<IConfigureOptions<McpServerOptions>, McpServerOptionsSetup>());
```

Comment:
The goal here is that you can call AddMcpServer multiple times and it's idempotent? I know you mentioned considering adding a test, but we probably should have a test somewhere that calls it multiple times if that's indeed something that should be supported?

Reply:
Exactly. I'll add a test. Even though it wouldn't fail before this change since McpServerOptionsSetup.Configure just happens to be idempotent today, a test could still catch future regressions.